pkg/cache/cache.go

```diff
@@ -443,13 +446,22 @@ func (c *Cache) UpdateClusterQueue(cq *kueue.ClusterQueue) error {
 func (c *Cache) DeleteClusterQueue(cq *kueue.ClusterQueue) {
 	c.Lock()
 	defer c.Unlock()
+	ctx := context.TODO()
+	logger := ctrl.LoggerFrom(ctx)
```

Comment:
```suggestion
func (c *Cache) DeleteClusterQueue(ctx context.Context, cq *kueue.ClusterQueue) {
	c.Lock()
	defer c.Unlock()
	logger := ctrl.LoggerFrom(ctx)
```

We should use propagated context.

Reply:
@kannon92 asked to save the propogated context for another PR to keep this one manageable

Reply:
I do not think so. If you want to avoid adding context in this PR. I would suggest preparation PR before this PR.
But, I am doubting that we do not need list lqs here. So, we might not use context in this function.

Reply:
If @mimowo and @tenzen-y prefer the context work in this PR, I am happy with their decision.